boa/explorer.py

```diff
@@ -12,13 +12,30 @@ def fetch_abi_from_etherscan(
 ):
     address = Address(address)
 
-    params = dict(module="contract", action="getabi", address=address)
+    params = dict(module="contract", action="getsourcecode", address=address)
```

Comment:
If this is only called to get the address, let's create a function `_get_implementation_address` so the purpose can be more clear